hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/store/accessor/TokenDatabaseAccessor.java

```
@@ -16,22 +16,102 @@
+                Collections.emptyList(),
+                Collections.emptyList(),
+                Collections.emptyMap(),
+                false,
+                Optional.ofNullable(databaseToken.getType())
+                        .map(t -> TokenType.valueOf(t.name()))
+                        .orElse(null),
+                Optional.ofNullable(databaseToken.getSupplyType())
+                        .map(st -> TokenSupplyType.valueOf(st.name()))
+                        .orElse(null),
+                Optional.ofNullable(databaseToken.getTotalSupply()).orElse(0L),
+                databaseToken.getMaxSupply(),
+                parseJkey(databaseToken.getKycKey()),
+                parseJkey(databaseToken.getFreezeKey()),
+                parseJkey(databaseToken.getSupplyKey()),
+                parseJkey(databaseToken.getWipeKey()),
+                parseJkey(entity.getKey()),
+                parseJkey(databaseToken.getFeeScheduleKey()),
+                parseJkey(databaseToken.getPauseKey()),
+                Boolean.TRUE.equals(databaseToken.getFreezeDefault()),
+                getTreasury(databaseToken.getTreasuryAccountId()),
+                null,
+                Optional.ofNullable(entity.getDeleted()).orElse(false),
+                TokenPauseStatusEnum.PAUSED.equals(databaseToken.getPauseStatus()),
+                false,
+                getEntityExpiration(entity),
+                false,
+                entity.getMemo(),
+                databaseToken.getName(),
+                databaseToken.getSymbol(),
+                Optional.ofNullable(databaseToken.getDecimals()).orElse(0),
+                Optional.ofNullable(entity.getAutoRenewPeriod()).orElse(0L),
+                0L);
+    }
+
+    private JKey parseJkey(byte[] keyBytes) {
+        try {
+            return keyBytes == null ? null : asFcKeyUnchecked(Key.parseFrom(keyBytes));
+        } catch (InvalidProtocolBufferException e) {
+            return null;
```

What kinds of problems throw `InvalidProtocolBufferException` here such that `null` is an acceptable value to provide the new `Token`?  Because I note that `asFcKeyUnchecked` is happy to throw `IllegalArgumentException`, not caught here and replaced with `null`, in case of `DecoderException`.  So there's a difference here between some errors that get silently replace by null and some that throw.

---

IllegalArgumentException added to catch.